spec/factories/loans.rb

```diff
@@ -55,11 +55,12 @@
 
 
     trait :active do
-      status_option_id Loan::STATUS_OPTIONS.value_for('Active')
+      # note: can't seem to count on the seed.db data having been loaded when these are executed
+      status_option_id 1 #Loan.status_option_set.value_for_label('Active')
```

Comment:
From my perspective there are two flavors of seed data.  And it's important to distinguish them.  Some of the data, like the `Currency`s and `Country`s are simply the most convenient way to handle the data migration.  It's much easier to insert a few static rows of data instead of writing business logic to migrate those from the old database.  (Building on my suggestion above, we can consider adding a `migration_id` column and remove any dependency on the primary key.  This is cleaner conceptually, but does add some complexity which I'm not entirely sure is justified, and is really a separate discussion from the thread here)  I completely agree that there should be no unit test or code dependency on data created for migration purposes that has no other specially business logic dependency.

But when there is data which is integrally tied to business logic, then I think it is legitimate for the unit tests to expect and depend on that data already existing.  I think this applies to the "active" and "completed" loan statuses.  The "root" division, a default Language (because the default translatable behavior fails with out that), and possibly a few of the other option sets like `public_level` if we have code dependencies on the "featured" and "hidden" values.

I think this set of data can be kept very minimal, left in seeds.rb and be automatically included automatically through the `spec/rails_helper.rb` logic.

Any other data (which includes the majority of what is currently in `seeds.rb` should be moved to a separate file and invoked as part of the data migration.

I think this balances things out.  We avoid the undesirable bloating of the seeds.rb as my current approach was leading us toward, but we avoid unnecessarily adding complexity and duplication to the unit test logic when there legitimate dependency on system data.  In this case, it feels more appropriate and simpler for the unit test to verify that the truly system dependent data was properly created by the seeds.rb than to duplicated its creation.  This last point is a personal preference and I don't feel too strongly about if you really think that the creation of that data should be duplicated when needed by various parts of the unit test code.